ledger/simulation/simulation_eval_test.go

```diff
@@ -2190,3 +2226,387 @@ func TestMockTracerScenarios(t *testing.T) {
 		})
 	}
 }
+
+func TestUnlimitedResourceAccessDisabled(t *testing.T) {
+	partitiontest.PartitionTest(t)
+	t.Parallel()
+
+	env := simulationtesting.PrepareSimulatorTest(t)
+	defer env.Close()
+	nodeConfig := config.GetDefaultLocal()
+	require.False(t, nodeConfig.EnableSimulationUnlimitedResourceAccess)
+	s := simulation.MakeSimulator(env.Ledger, nodeConfig)
+
+	sender := env.Accounts[0]
+
+	txn := env.TxnInfo.NewTxn(txntest.Txn{
+		Type:     protocol.PaymentTx,
+		Sender:   sender.Addr,
+		Receiver: sender.Addr,
+	})
+	stxn := txn.Txn().Sign(sender.Sk)
+
+	_, err := s.Simulate(
+		simulation.Request{
+			TxnGroups:                    [][]transactions.SignedTxn{{stxn}},
+			AllowUnlimitedResourceAccess: true,
+		})
+	require.ErrorAs(t, err, &simulation.InvalidRequestError{})
+	require.ErrorContains(t, err, "unlimited resource access is not enabled in node configuration: EnableSimulationUnlimitedResourceAccess is false")
+}
+
+// TestUnlimitedResourceAccess tests that app calls can access resources that they otherwise
+// should not be able to if AllowUnlimitedResourceAccess is enabled. Additional tests follow for
+// special cases.
+func TestUnlimitedResourceAccess(t *testing.T) {
```

Comment:
This test looks a bit crazy, but it is reasonable to test across all versions here.